File: tests/python/arrays/line_through_points/test_line_through_points.py
from problems.arrays.line_through_points.python.line_through_points import \
    lineThroughPoints
from tests.python.utility import load_test_cases


def test_line_through_points() -> None:
    test_cases = load_test_cases("arrays/line_through_points.json")

    for idx, test in enumerate(test_cases):
        points: list[list[int]] = test["points"]
        expected: int = test["expected"]

        result: int = lineThroughPoints(points)

        assert result == expected, f"Test case {idx} failed"

File: tests/cpp/arrays/test_line_through_points.cpp
#include "../../../problems/arrays/line_through_points/cpp/line_through_points.h"
#include "../jsontestbase.h"

#include <vector>

class LineThroughPointsTest : public JsonTestBase {
public:
  LineThroughPointsTest() {
    json_file_path =
        std::string(TEST_CASES_DIR) + "/arrays/line_through_points.json";
  }
};

TEST_F(LineThroughPointsTest, TestCases) {
  ASSERT_FALSE(test_cases.empty()) << "Test cases are empty or not loaded";

  for (const auto &test : test_cases) {
    auto points = test["points"].get<std::vector<std::vector<int>>>();
    auto expected = test["expected"].get<int>();

    auto result = lineThroughPoints(points);

    EXPECT_EQ(result, expected) << "Failed on test case: " << test.dump(2);
  }
}
